sipa/utils/__init__.py

```diff
@@ -82,6 +87,37 @@ def support_hotline_available():
         return False
 
 
+def meetingcal():
+    url = "https://agdsn.de/cloud/remote.php/dav/public-calendars/bgiQmBstmfzRdMeH?export"
+
+    try:
+        response = requests.get(url, timeout=1)
+    except requests.exceptions.RequestException:
+        return []
+        
+    if response.status_code != 200:
+        return []
+    
+    try:
+        calendar = icalendar.Calendar.from_ical(response.text)
+    except ValueError:
+        return []
+
+    events = recurring_ical_events.of(calendar).between(datetime.now(), datetime.now() + relativedelta(months=1))
+
+    next_meetings = []
+    for event in events:
+        next_meetings.append({
+            "title": event["SUMMARY"],
+            "datetime": event["DTSTART"].dt,
+            "location": event["LOCATION"] if "LOCATION" in event else "-",
+            "location_link": markdown.markdown(event["LOCATION"]) if "LOCATION" in event else "-"
+        })
+
+    next_meetings = sorted(next_meetings, key=lambda x: x["datetime"])
+
+    return next_meetings
```

Comment:
I forgot that it is really bad to make a synchronous HTTP request every time, so this whole function should be cached. Two options that you could use:

- A session based cache like it is used in `support_hotline_available()`
- Using [cachetools](https://pypi.org/project/cachetools/) and decorating the function with `
@cachetools.func.ttl_cache(maxsize=1, ttl=5 * 60)`